### PR TITLE
style: Pass at propagating (and enhancing) Button component throughout Superset

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY

image

There are a LOT of ways to make a button on the web. With this PR merged, there will be ONE component used to replace them all (or so I _believe_).

This PR does a number of little things:
• Replaces any direct use of the `react-bootstrap` Button component with the one from `/components/Button`
• Gets rid of a blue 'outline' on focus/active state of buttons.
• Makes the "published status" of a dashboard into a clickable label, rather than the dotted-outline rectangle button
• replaces HTML `<button>`s with `Button`
• removes a `.text-button` class, since there's a prop for that use case in the Button component
• abstracts some bootstrap props to make a better "pivot point" component, e.g. `bsStyle` --> `buttonStyle`, `bsSize` --> `buttonSize`
• adds a `cta` prop (call to action) for "important" buttons that are used in things like modals, and an increasing number of new views. These are basically wider, with a fixed width. See Storybook for examples
• adds secondary/tertiary/dashed button styles from new design system (see Storybook)
• ... and more?

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Outline issue before/after
before
after

This change is actually a Label, but worth noting it has a visual impact.
published

Full size buttons on all(?) Modals
image

New "New" button on global menu:
image

These buttons don't look great, but that gray background behind them will become white soon in a subsequent PR, which will look much better.
image

### TEST PLAN

Several tests tweaked to work. Button unit tests WIP.

### ADDITIONAL INFORMATION

- [x] Has associated issue: #8976
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
image
More new buttons in popovers.

**Comment by rusackas** [comment]
There are probably a number of little things left to touch up, but I'd like to address those in another PR. This one is getting WAY too big. Thanks to anyone that can take a look at this and make sure nothing is causing any major issues.

File: superset-frontend/src/views/CRUD/data/dataset/Button.tsx [hunk 1]
```diff
@@ -1,72 +0,0 @@
-/**
```
Comment (nytai) [inline superset-frontend/src/views/CRUD/data/dataset/Button.tsx:1]: 🙌 🥳 🎉

File: superset-frontend/spec/helpers/theming.ts [hunk 2]
```diff
@@ -51,5 +51,5 @@ export function styledShallow(
       theme: supersetTheme,
       ...options?.wrappingComponentProps,
     },
-  });
+  }).dive();
```
Comment (nytai) [inline superset-frontend/spec/helpers/theming.ts:54]: Should this live here or in the tests using this wrapper? I see it called here: https://github.com/apache/incubator-superset/pull/10649/files#diff-7dabcc63260bfc89250effddae3e73d5R49

are both necessary?

**Review by nytai — APPROVED** [review]
lgtm, just one question

File: superset-frontend/spec/helpers/theming.ts (hunk 2 above)
Comment (rusackas) [reply superset-frontend/spec/helpers/theming.ts:54]: Before moving the `dive()` into the helper, That instance had `dive().dive()` happening on it. I can't think of a time when we wouldn't want to `dive()` at least once, so I added it into the helper to reduce/reuse/recycle.

Comment (nytai) [reply superset-frontend/spec/helpers/theming.ts:54]: ah, so dive only goes 1 component deep, makes sense

**Pull request merged**